### PR TITLE
Fix NullReferenceException when serializing null record-typed record field

### LINKED ISSUE #172: Regression in 1.2: NullReferenceException when serializing null record field
Repro:

```f#
open System
open System.Text.Json
open System.Text.Json.Serialization

type X = { X: X }

let x = { X = Unchecked.defaultof<_> }

let options = JsonSerializerOptions()
options.Converters.Add(JsonFSharpOptions.Default() |> JsonFSharpConverter)

Console.WriteLine(JsonSerializer.Serialize(x, options))
```

On v1.1 this works fine, but on v1.2 this throws:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at lambda_method2(Closure, Object)
   at System.Text.Json.Serialization.JsonRecordConverter`1.WriteRestOfObject(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonRecordConverter`1.Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCoreAsObject(Utf8JsonWriter writer, Object value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCoreAsObject(Utf8JsonWriter writer, Object value, JsonTypeInfo jsonTypeInfo)
   at System.Text.Json.Serialization.JsonRecordConverter`1.WriteRestOfObject(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonRecordConverter`1.Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore[TValue](Utf8JsonWriter writer, TValue& value, JsonTypeInfo`1 jsonTypeInfo)
   at System.Text.Json.JsonSerializer.WriteString[TValue](TValue& value, JsonTypeInfo`1 jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at <StartupCode$Test2>.$Program.main@() in \_\Test2\Test2\Program.fs:line 12
```

### PR DESCRIPTION
**Pull request by cmeeren**
Fixes #172

### CODE REVIEW

**Comment by cmeeren** [comment]
I think this is good to go. Please review. Would be great if it could be released quickly, since the bug is causing some headaches in prod right now.

File: src/FSharp.SystemTextJson/Record.fs [hunk 1]
```diff
@@ -205,8 +205,11 @@ type JsonRecordConverter<'T> internal (options: JsonSerializerOptions, fsOptions
         ctor fields :?> 'T
 
     override this.Write(writer, value, options) =
-        writer.WriteStartObject()
-        this.WriteRestOfObject(writer, value, options)
+        if isNull (box value) then
+            JsonSerializer.Serialize(writer, null, options)
```
Comment (Tarmil) [inline src/FSharp.SystemTextJson/Record.fs:209]: I think `writer.WriteNullValue()` would be both clearer and more efficient.

**Review by Tarmil — CHANGES_REQUESTED** [review]

File: src/FSharp.SystemTextJson/Record.fs (hunk 1 above)
Comment (cmeeren) [reply src/FSharp.SystemTextJson/Record.fs]: Agreed, done

**Review by Tarmil — APPROVED** [review]

**Comment by Tarmil** [comment]
I'm going to rebase this onto the branch `releases/v1.2` for a patch release.

**Comment by cmeeren** [comment]
Great, thanks!

**Comment by Tarmil** [comment]
Close in favor of #174.

**Pull request closed without merging**